x-pack/plugins/observability_solution/slo/server/services/get_slo_health.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TransformGetTransformStatsTransformStats } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient } from '@kbn/core/server';
+import {
+  FetchSLOHealthParams,
+  FetchSLOHealthResponse,
+  fetchSLOHealthResponseSchema,
+} from '@kbn/slo-schema';
+import { Dictionary, groupBy, keyBy } from 'lodash';
+import moment from 'moment';
+import {
+  getSLOSummaryTransformId,
+  getSLOTransformId,
+  SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+} from '../../common/constants';
+import { SLODefinition } from '../domain/models';
+import { HealthStatus, State } from '../domain/models/health';
+import { SLORepository } from './slo_repository';
+import { EsSummaryDocument } from './summary_transform_generator/helpers/create_temp_summary';
+
+const LAG_THRESHOLD_MINUTES = 10;
+const STALE_THRESHOLD_MINUTES = 2 * 24 * 60;
+
+export class GetSLOHealth {
+  constructor(private esClient: ElasticsearchClient, private repository: SLORepository) {}
+
+  public async execute(params: FetchSLOHealthParams): Promise<FetchSLOHealthResponse> {
+    const sloIds = params.list.map(({ sloId }) => sloId);
+    const sloList = await this.repository.findAllByIds(sloIds);
+    const sloById = keyBy(sloList, (slo) => slo.id);
+
+    const filteredList = params.list
+      .filter((item) => !!sloById[item.sloId])
+      .map((item) => ({
+        sloId: item.sloId,
+        sloInstanceId: item.sloInstanceId,
+        sloRevision: sloById[item.sloId].revision,
+      }));
+
+    const transformStatsById = await this.getTransformStats(sloList);
+    const summaryDocsById = await this.getSummaryDocsById(filteredList);
+
+    const results = filteredList.map((item) => {
+      const health = computeHealth(transformStatsById, item);
+      const state = computeState(summaryDocsById, item);
+
+      return {
+        sloId: item.sloId,
+        sloInstanceId: item.sloInstanceId,
+        sloRevision: item.sloRevision,
+        state,
+        health,
+      };
+    });
+
+    return fetchSLOHealthResponseSchema.encode(results);
+  }
+
+  private async getSummaryDocsById(
+    filteredList: Array<{ sloId: string; sloInstanceId: string; sloRevision: number }>
+  ) {
+    const summaryDocs = await this.esClient.search<EsSummaryDocument>({
+      index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
```

Comment:
will this handle remote SLOs?

Reply:
No because transforms health cannot be fetch remotely anyway